storage/localstore/gc.go

```diff
@@ -106,6 +108,13 @@ func (db *DB) collectGarbage() (collectedCount uint64, done bool, err error) {
 			return true, nil
 		}
 
+		// Check if this chunk is pinned, if YES, then skip GC on that chunk
+		pinItem, err := db.pinIndex.Get(item)
```

Comment:
pinned chunks should probably not show up in the `gc` index in the first place. this might incur a lot of unnecessary i/o that can be prevented

Reply:
Well this is a design decision i took.  If you store the pinned chunks (with data) in a separate index other than retrievalDataIndex, then every time you pin or unpin a file.. you incur heavy cost of moving the data chunk to another index. If we keep a separate data chunk index for pinning, then all the other things like sync should take care of two indexes...which is clumsy.

But if you store data in the same index and just play with the address using another index.. the write cost is very very low.

Also in levelDB read is faster.. so you should not see a big drop in performance while doing GC.

Reply:
Done.